svd-rs/src/writeconstraint.rs

```diff
@@ -25,3 +27,28 @@ pub struct WriteConstraintRange {
     #[cfg_attr(feature = "serde", serde(rename = "maximum"))]
     pub max: u64,
 }
+
+/// Errors for [`WriteConstraintRange::check_range`]
+#[derive(Clone, Debug, PartialEq, Eq, thiserror::Error)]
+pub enum Error {
+    /// The value is not in range.
+    #[error("Value {0} out of range {1:?}")]
+    OutOfRange(u64, core::ops::Range<u64>),
+    /// Minimum is greater than maximum.
+    #[error("Range minimum {0} is greater than maximum {1}")]
+    ReversedRange(u64, u64),
+}
+
+impl WriteConstraintRange {
+    pub(crate) fn check_range(&self, range: core::ops::Range<u64>) -> Result<(), SvdError> {
+        if &self.min > &self.max {
```

Comment:
I don't see sense in references here. Could you run `cargo clippy`, please.
Looks like I missed it in CI.

Reply:
Yeah, you right